app/processing/index.js

```diff
@@ -1,20 +1,17 @@
 const { processingConfig } = require('../config')
```

Comment:
I think `transaction` should live in this file because it acts as the start/finish of this transaction (sort of like a `main` method)

To rectify the problem we just need to call `transaction.commit` on line 20 if all is well and `transaction.rollback` in the catch if not.

I can see tht `sendStatement` is also missing a `transaction`param

What do you think?

Reply:
There is no direct or next call of database in index.js, transaction management may become an issue when called in-appropriately at parent-level, hence, the reason we were initially having ConnectionAcquireTimeoutError.

Reply:
"There is no direct or next call of database in index.js" that was correct, it was missing -- we started a `transaction` on line 9 previously but never committed it, hence the timeout issues. If we re-instate the `transaction` on line 9 again and `transaction.commit` on line 19 and `transaction.rollback` on line 21

I strongly feel this is where the transaction should be managed not within the individually called funcs. I can see `transaction.commit` is called within `schedulePendingSettlements` and `getStatement`. These are called within here (first `schedulePendingSettlements` and then `getStatement`) which posses a problem.

If `schedulePendingSettlements` acts on 3 new records: performing gets, updates and save downs (it might not be but bare with) and completes successfully so calls `transaction.commit` then `getStatement` is called acting on the 3 records: however, it fails so it calls `transaction.rollback`

Becuz `schedulePendingSettlements` committed and `getStatement` rollbacked the DB is not consistent anymore

Reply:
The implication of what you said above is that if there are 100 statements in a `schedulePendingSettlements ` if there is an issue with one of the settlements, then we should not send any of the 100 statements. Could you please confirm this? In my opinion, I don't think we should block all statements in a schedule because of one or few having problem, this is similar to the issue we had with paymentRequest validator in the past when we were blocking a whole batch from being validated because of one or few invalid paymentRequest file. I think we should adopt the same approach we used for paymentRequest validator.

Reply:
V valid point.

I don't believe the current setup (or mine!) will enable partial processing of requests if 1 fails within a batch but as you mentioned defo sommit to add in. Happy to approve as is, providing we add this ticket to the backlog.